### PR TITLE
Blocked classic prometheus constructors, moved all to promauto; Removed unnecessary printfs.

### PR DESCRIPTION
**Pull request by bwplotka**
This is thanks to https://github.com/fatih/faillint/issues/7

### CODE REVIEW

File: pkg/alert/alert.go [hunk 1]
```diff
@@ -133,34 +134,31 @@ func NewQueue(logger log.Logger, reg prometheus.Registerer, capacity, maxBatchSi
...
 			Help: "Total number of alerts pushed to the queue.",
 		}),
-		popped: prometheus.NewCounter(prometheus.CounterOpts{
+		popped: promauto.With(reg).NewCounter(prometheus.CounterOpts{
 			Name: "thanos_alert_queue_alerts_popped_total",
 			Help: "Total number of alerts popped from the queue.",
 		}),
 	}
-	capMetric := prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+	_ = promauto.With(reg).NewGaugeFunc(prometheus.GaugeOpts{
```
Comment (squat) [inline pkg/alert/alert.go]: Is it better to do `_ = func()` or just `func()`?

Comment (bwplotka) [reply pkg/alert/alert.go]: Well I do `_` to indicate that this returns value, but we don't need it. Do you think it's wrong pattern?

Comment (bwplotka) [reply pkg/alert/alert.go]: I like it as it avoid surprises (: But happy to change this in the following PRs :hugs:

Comment (squat) [reply pkg/alert/alert.go]: I just wanted to make sure we were being consistent in the project one way or the other so :+1:

**Pull request merged**